model/config.go

```diff
@@ -1449,6 +1450,28 @@ func (s *FileSettings) SetDefaults(isUpdate bool) {
 	}
 }
 
+func (s *FileSettings) ToFileBackendSettings(enableComplianceFeature bool) filesstore.FileBackendSettings {
+	if *s.DriverName == IMAGE_DRIVER_LOCAL {
+		return filesstore.FileBackendSettings{
+			DriverName: *s.DriverName,
+			Directory:  *s.Directory,
+		}
+	}
+	return filesstore.FileBackendSettings{
+		DriverName:              *s.DriverName,
+		AmazonS3AccessKeyId:     *s.AmazonS3AccessKeyId,
+		AmazonS3SecretAccessKey: *s.AmazonS3SecretAccessKey,
+		AmazonS3Bucket:          *s.AmazonS3Bucket,
+		AmazonS3PathPrefix:      *s.AmazonS3PathPrefix,
+		AmazonS3Region:          *s.AmazonS3Region,
+		AmazonS3Endpoint:        *s.AmazonS3Endpoint,
+		AmazonS3SSL:             s.AmazonS3SSL == nil || *s.AmazonS3SSL,
+		AmazonS3SignV2:          s.AmazonS3SignV2 != nil && *s.AmazonS3SignV2,
+		AmazonS3SSE:             s.AmazonS3SSE != nil && *s.AmazonS3SSE && enableComplianceFeature,
+		AmazonS3Trace:           s.AmazonS3Trace != nil && *s.AmazonS3Trace,
```

Comment:
Wondering if we should converge on either assuming the fields are not `nil` or check for `nil` on all of them. As it is it seems arbitrary (except maybe for `AmazonS3SSL` which defaults to `true`).

Reply:
It is based on the pre-existing code, we can check everything there, but that, in theory is checked before save, in theory some of that fields can't be nil (can be empty string, or a value, but the pointer is to allow config.json to not-have the field). Once is in the system we expect some fields to be set. Probably we should ensure all of them and remove the nil check for everything else. (Out of the scope of this PR, a good thing to decide and do all over the code).